lib.php

```diff
@@ -281,9 +294,20 @@ abstract public function get_settings(block_base $block);
      */
     abstract public function validate_dates(block_base $block, array $dates);
 
+    /**
+     * Save the new dates for this block instance.
+     *
+     * @param \block_base $block the block to save the dates for.
+     * @param array $dates a list of new dates.
+     */
+    final public function save_new_dates(block_base $block, array $dates) {
```

Comment:
There is no real reason to do have this new method for for the block instance. It's done for consistency.

Reply:
I'm not too sure what you mean by consistence here. Consistency with what?

Reply:
So we added save_new_dates method to modules related class, hence I'd like to keep it consistent for blocks as well. But technically there is no reason to do so as we don't trigger anything for the blocks.

Reply:
If even Cameron does not understand what is giong on here, then you need to improve the comment (instead of answering in  github comment which will neve be seen again.)

You also need to write your pull request to explain what you are doing and why. Yiou may konw, but I am not telepathic.

Reply:
I have updated a pull request description to explain what's happening in the code. As well as add comment to explain a new method a little bit more. Please have a look and let me know if that's better now.